logic/service/raw-data-service.js

```diff
@@ -36,6 +40,19 @@ exports.RawDataService = DataService.specialize(/** @lends RawDataService.protot
         }
     },
 
+    initWithModel: {
```

Comment:
First, we need a setter for model, second what's passed here as an argument is model's module id.

Also, it would be unusual for an init method to return a promise I think